src/System.Reflection.Metadata/src/System/Reflection/Metadata/Ecma335/MetadataBuilder.Heaps.cs

```diff
@@ -290,14 +290,14 @@ internal void CompleteHeaps()
 
         public ImmutableArray<int> GetHeapSizes()
         {
-            var heapSizes = new int[MetadataTokens.HeapCount];
+            var builder = ImmutableArray.CreateBuilder<int>(MetadataTokens.HeapCount);
 
-            heapSizes[(int)HeapIndex.UserString] = _userStringWriter.Count;
-            heapSizes[(int)HeapIndex.String] = _stringWriter.Count;
-            heapSizes[(int)HeapIndex.Blob] = _blobHeapSize;
-            heapSizes[(int)HeapIndex.Guid] = _guidWriter.Count;
+            builder[(int)HeapIndex.UserString] = _userStringWriter.Count;
```

Comment:
@tmat I hate to disagree, but these are public methods and the changes are relatively trivial. I'll close this for now, though. :smile: